flink/src/main/scala/ru/itclover/tsp/PatternsSearchJob.scala

```diff
@@ -105,20 +110,30 @@ case class PatternsSearchJob[In: TypeInformation, InKey, InItem](
           source.emptyEvent
         )(timeExtractor) //.asInstanceOf[StatefulFlatMapper[In, S, Incident]]
     }
-    val res = stream
-      .assignAscendingTimestamps(timeExtractor(_).toMillis)
-      .keyBy(source.partitioner)
-      .window(
-        TumblingEventTimeWindows
-          .of(WindowingTime.milliseconds(source.conf.chunkSizeMs.getOrElse(900000)))
-          .asInstanceOf[WindowAssigner[In, FlinkWindow]]
-      )
-//      .window(GlobalWindows.create().asInstanceOf[WindowAssigner[In, FlinkWindow]])
-//      .trigger(EventCounterTrigger[In, FlinkWindow](source.conf.chunkSize.getOrElse(5000)))
-      .process[Incident](
-        ProcessorCombinator[In, S, Segment, Incident](mappers)
-      )
-      .setMaxParallelism(source.conf.maxPartitionsParallelism)
+    val res =  if (useWindowing) {
```

Comment:
Please, extract common parts of those branches to make difference more obvious. Smth like:
```
val keyedStream = stream.assign...().keyBy(...)
val windowed = if(useWIndowing) {...} else {...}
val processedStream = windowed.process(...).setMaxParallelism(...)
```

Reply:
Fixed